lib/posix/options/pthread.c

```diff
@@ -869,6 +869,46 @@ int pthread_setschedparam(pthread_t pthread, int policy, const struct sched_para
 	return ret;
 }
 
+/**
+ * @brief Set thread scheduling priority.
+ *
+ * See IEEE 1003.1
+ */
+int pthread_setschedprio(pthread_t thread, int prio)
+{
+	int ret;
+	int new_prio = K_LOWEST_APPLICATION_THREAD_PRIO;
+	struct posix_thread *t = NULL;
+	int policy;
+	struct sched_param param;
+
+	ret = pthread_getschedparam(thread, &policy, &param);
+
+	if (ret != 0) {
+		return ret;
+	}
```

Comment:
This should be removed because it creates a race condition.

Reply:
Actually nevermind. It's not a good idea to call when inside of a spin lock..

It is somewhat racey...